### ISSUE TITLE
crash with batchSize = 2 and GPT2-WikiText2 demo

### ISSUE THREAD

**Issue by brettkoonce** [tensorflow/swift-models#374]
set batchSize to 2 in the demo, add following to loop over dataset:

```
import Batcher
import Datasets
import TensorFlow
import TextModels

var gpt = try GPT2()

let sequenceLength = gpt.contextSize
let batchSize = 2
let numWorkers = 8
// Use default WikiText2 dataset.
let dataset = TextUnsupervised(bpe: gpt.bpe, variant: .wikiText2, batchSize: batchSize,
    sequenceLength: sequenceLength)
let trainingBatcher = Batcher(
    on: dataset.trainingDataset, batchSize: batchSize, numWorkers: numWorkers, shuffle: true)
let validationBatcher = Batcher(
    on: dataset.validationDataset, batchSize: batchSize, numWorkers: numWorkers)

print("Dataset acquired.")

var optimizer = Adam(for: gpt.model, learningRate: 0.001)

for batch in trainingBatcher.sequenced() {
    let (documents, labels) = (batch.first, batch.second)
}

print("Starting training...")
```

run: `swift run GPT2-WikiText2`

Fatal error: Shapes of all inputs must match: values[0].shape = [1024] != values
[1].shape = [500]: file /home/skoonce/swift/swift-source/tensorflow-swift-apis/S
ources/TensorFlow/Bindings/EagerExecution.swift, line 300                       

works with bs = 1, something in https://github.com/tensorflow/swift-models/blob/b4c8b0e5b591bef13858cb89b4eec9d6de90bceb/Batcher/Batcher.swift#L108
and that the wikitext data isn't uniform is tripping this, looks like.

ping @sgugger

**Comment by sgugger** [comment]
@texasmichelle was reporting something similar when changing `sequenceLength` from 72 to 1024. I'll investigate tomorrow and see if I can reproduce.

**Comment by texasmichelle** [comment]
Thanks for surfacing this issue! I'll take a look.

**Comment by brettkoonce** [comment]
@texasmichelle tis a cool demo, was trying not to spam you with nits!

**Comment by texasmichelle** [comment]
Thanks for the quick fix @sgugger ! I added a few comments since it fixes the batchSize = 2 case, but breaks the default batch size of 64. A bit of additional troubleshooting info: without your fix, if I inject a shape check that skips batches with lengths less than sequenceLength, this works fine as long as I have shuffling turned off. With shuffling on, the error seems to occur prior to the shape check, potentially at the .sequenced() call.

@brettkoonce I appreciate the nits! I'm new to swift and it's a great way to learn. Please open more issues and add comments to my PRs 🙏
